src/test/java/io/confluent/connect/jdbc/sink/JdbcDbWriterTest.java

```diff
@@ -71,6 +80,73 @@ private JdbcDbWriter newWriter(Map<String, String> props) {
     return new JdbcDbWriter(config, dialect, dbStructure);
   }
 
+  private JdbcDbWriter newWriterWithMockConnection(Map<String, String> props, Connection mockConnection) {
+    final JdbcSinkConfig config = new JdbcSinkConfig(props);
+    dialect = mock(DatabaseDialect.class);
+    final DbStructure dbStructure = mock(DbStructure.class);
+    return new JdbcDbWriter(config, dialect, dbStructure) {
+      protected CachedConnectionProvider connectionProvider(int maxConnAttempts, long retryBackoff) {
+        CachedConnectionProvider mockConnectionProvider = mock(CachedConnectionProvider.class);
+        when(mockConnectionProvider.getConnection()).thenReturn(mockConnection);
+        return mockConnectionProvider;
+      }
+    };
+  }
+
+  @Test
+  public void verifyConnectionRollback() throws SQLException {
+    Connection mockConnection = mock(Connection.class);
+
+    doThrow(new SQLException()).when(mockConnection).commit();
+    doThrow(new SQLException()).when(mockConnection).rollback();
```

Comment:
Maybe throw a different exception on rollback to so we can be sure the suppressed exception is different from the exception thrown on `commit()` ?

Reply:
I created a fake exception class called `MockRollbackException` that extends `SQLException` to be sure we are getting the right suppressed exception. It needs to extend `SQLException` because the code (accurately) only catches `SQLException`s